Gems/Atom/RHI/DX12/Code/Source/RHI/StreamingImagePool.cpp

```diff
@@ -117,110 +127,178 @@ namespace AZ
+                heapPageAllocatorDesc.m_getHeapMemoryUsageFunction = [this]() { return &GetDeviceHeapMemoryUsage();};
+                heapPageAllocatorDesc.m_pageSizeInBytes = TileSizeInBytes * 256; // 16M per page, 256 tiles
+                heapPageAllocatorDesc.m_resourceTypeFlags = ResourceTypeFlags::Image; 
+                heapPageAllocatorDesc.m_heapMemoryLevel = RHI::HeapMemoryLevel::Device;
+                heapPageAllocatorDesc.m_hostMemoryAccess = RHI::HostMemoryAccess::Write;
+                heapPageAllocatorDesc.m_collectLatency = 0;
+                heapPageAllocatorDesc.m_recycleOnCollect = false;    // Release the heap page when the TileAllocator deallocate it
+
+                m_heapPageAllocator.Init(heapPageAllocatorDesc);
+
+                TileAllocator::Descriptor tileAllocatorDesc;
+                tileAllocatorDesc.m_tileSizeInBytes = TileSizeInBytes;
+                // Tile allocator updates used resident memory
+                tileAllocatorDesc.m_getHeapMemoryUsageFunction = heapPageAllocatorDesc.m_getHeapMemoryUsageFunction;
+                m_tileAllocator.Init(tileAllocatorDesc, m_heapPageAllocator);
+
+                // allocate one tile for default tile
+                auto heapTiles = m_tileAllocator.Allocate(1);
+                AZ_Assert(heapTiles.size() == 1, "Failed to allocate a default heap");
+                m_defaultTile = heapTiles[0];
             }
-#endif
 
             SetResolver(AZStd::make_unique<StreamingImagePoolResolver>());
             return RHI::ResultCode::Success;
         }
 
         void StreamingImagePool::ShutdownInternal()
         {
-#ifdef AZ_RHI_USE_TILED_RESOURCES
-            GetDevice().QueueForRelease(AZStd::move(m_heap));
-            m_tileAllocator.Shutdown();
-#endif
+            if (m_enableTileResource)
+            {
+                m_tileAllocator.Shutdown();
+            }
         }
 
-        void StreamingImagePool::AllocateImageTilesInternal(Image& image, CommandList::TileMapRequest& request, uint32_t subresourceIndex)
+        void StreamingImagePool::AllocateImageTilesInternal(Image& image, uint32_t subresourceIndex)
         {
-            AZ_Assert(request.m_destinationHeap, "Destination heap must be valid.");
-
             uint32_t imageTileOffset = 0;
+            CommandList::TileMapRequest request;
+            request.m_sourceMemory = image.GetMemoryView().GetMemory();
             image.m_tileLayout.GetSubresourceTileInfo(subresourceIndex, imageTileOffset, request.m_sourceCoordinate, request.m_sourceRegionSize);
 
-            request.m_destinationTileMap.resize(request.m_sourceRegionSize.NumTiles);
+            AZ_Assert(image.m_heapTiles[subresourceIndex].size() == 0, "Stoping on an existing tile allocation. This will leak.");
+
+            uint32_t totalTiles = request.m_sourceRegionSize.NumTiles;
+            image.m_heapTiles[subresourceIndex] = m_tileAllocator.Allocate(totalTiles);
```

Comment:
Is there something preventing this from being called from multiple threads at the same time?

Reply:
Yes, we have a mutex at where the AllocateImageTilesInternal were called.